app/components/autocomplete/autocomplete.tsx

```diff
@@ -0,0 +1,202 @@
+        },
+        borders: {
+            borderWidth: 1,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.2),
+            overflow: 'hidden',
+            borderRadius: 4,
+        },
+        hidden: {
+            display: 'none',
+        },
+        searchContainer: {
+            ...Platform.select({
+                android: {
+                    top: 42,
+                },
+                ios: {
+                    top: 55,
+                },
+            }),
+        },
+        shadow: {
+            shadowColor: '#000',
+            shadowOpacity: 0.12,
+            shadowRadius: 8,
+            shadowOffset: {
+                width: 0,
+                height: 8,
+            },
+        },
+    };
+});
+
+type Props = {
+    cursorPosition: number;
+    maxHeight: number;
+    rootId: string;
+    channelId: string;
+    isSearch?: boolean;
+    value: string;
+    enableDateSuggestion?: boolean;
+    appsEnabled: boolean;
+    offsetY?: number;
+    nestedScrollEnabled?: boolean;
+    updateValue: (v: string) => void;
+}
+const Autocomplete = ({
+    cursorPosition,
+    maxHeight,
+    rootId,
+
+    //channelId,
+    isSearch = false,
+    value,
+
+    //enableDateSuggestion = false,
+    appsEnabled,
+    offsetY = 80,
+    nestedScrollEnabled = false,
+    updateValue,
+}: Props) => {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+    const dimensions = useWindowDimensions();
+    const deviceHeight = dimensions.height;
+
+    // const [showingAtMention, setShowingAtMention] = useState(false);
+    // const [showingChannelMention, setShowingChannelMention] = useState(false);
+    const [showingEmoji, setShowingEmoji] = useState(false);
+
+    // const [showingCommand, setShowingCommand] = useState(false);
+    // const [showingAppCommand, setSwhoingAppCommand] = useState(false);
+    // const [showingDate, setShowingDate] = useState(false);
+
+    const hasElements = showingEmoji; // || showingAtMention || showingChannelMention || showingCommand || showingAppCommand || showingDate;
+    const appsTakeOver = false; // showingAppCommand;
+
+    const maxListHeight = useMemo(() => {
+        if (maxHeight) {
+            return maxHeight;
+        }
+
+        // List is expanding downwards, likely from the search box
+        let offset = Platform.select({ios: 65, default: 75});
+        if (Device.IS_IPHONE_WITH_INSETS) {
+            offset = 90;
+        }
+
+        return (deviceHeight / 2) - offset;
+    }, [maxHeight, deviceHeight]);
+
+    const wrapperStyles = useMemo(() => {
+        const s = [];
+        if (Platform.OS === 'ios') {
+            s.push(style.shadow);
+        }
+
+        if (isSearch) {
+            s.push(style.base, style.searchContainer, {height: maxListHeight});
+        }
+
+        if (!hasElements) {
+            s.push(style.hidden);
+        }
+        return s;
+    }, [theme, isSearch, hasElements, maxListHeight]);
```

Comment:
Should you have `style` here instead of `theme`?  Although, it's when the theme changes that the style changes.